manifest/manifest.go

```diff
@@ -54,7 +54,7 @@ func GuessMIMEType(manifest []byte) string {
 	}
 
 	switch meta.MediaType {
-	case DockerV2Schema2MediaType, DockerV2ListMediaType, imgspecv1.MediaTypeImageManifest, imgspecv1.MediaTypeImageManifestList: // A recognized type.
+	case DockerV2Schema2MediaType, DockerV2ListMediaType: // A recognized type.
```

Comment:
Either way, this is awkward.

* `dirImageSource` and `storageImageSource` depend on `GuessMIMEType` to work; otherwise `c/i/image` assumes schema1. (Not _trivially_ fixable.)
* `dockerImageDestination` depends on `GuessMIMEType` for setting `Content-Type` on upload. (This one is probably fixable by adding an explicit parameter, and using the `c/i/image` handler to get the type we are using.)

Is there another heuristic we can use? e.g. `schemaVersion` == `2` && `mediaType` missing? Or perhaps together with `config.mediaType`?